full/full_coupler_mod.F90

```diff
@@ -277,6 +274,41 @@ module full_coupler_mod
     integer :: flux_exchange_init
   end type coupler_clock_type
 
+  type coupler_components_type
+    private
+    type(atmos_data_type), pointer :: Atm  !< pointer to Atm
+    type(land_data_type),  pointer :: Land !< pointer to Land
+    type(ice_data_type),   pointer :: Ice  !< pointer to Ice
+    type(ocean_public_type), pointer :: Ocean  !< pointer to Ocean
+    type(land_ice_atmos_boundary_type), pointer :: Land_ice_atmos_boundary !< pointer to Land_ice_atmos_boundary
+    type(atmos_land_boundary_type), pointer :: Atmos_land_boundary !< pointer to Atmos_land_boundary
+    type(atmos_ice_boundary_type),  pointer :: Atmos_ice_boundary  !< pointer to Atmos_ice_boundary
+    type(land_ice_boundary_type),   pointer :: Land_ice_boundary   !< pointer to Land_ice_boundary
+    type(ice_ocean_boundary_type),  pointer :: Ice_ocean_boundary  !< pointer to Ice_ocean_boundary
+    type(ocean_ice_boundary_type),  pointer :: Ocean_ice_boundary  !< pointer to Ocean_ice_boundary
+  contains
+    procedure, public :: initialize_coupler_components_obj
+    procedure, public :: get_component  !< subroutine to retrieve the requested component of an object of this type
+    procedure, public :: set_component  !< subroutine to set requested component of an object of this type
+  end type coupler_components_type
+
+  !> The purpose of objects of coupler_chksum_type is to simplify the list
+  !! of arguments required for chksum related subroutines in full_coupler_mod.
+  !! The members of this type point to the model components
+  type coupler_chksum_type
+    private
+    type(coupler_components_type) :: components
```

Comment:
are these two of these?  Should this point to `coupler_components_obj`?

Reply:
it has been changed to a pointer